website/src/controllers/groups.js

```diff
@@ -518,10 +518,12 @@ api.leave = function(req, res, next) {
   // When removing the user from challenges, should we keep the tasks?
   var keep = (/^remove-all/i).test(req.query.keep) ? 'remove-all' : 'keep-all';
 
-  group.leave(user, keep, function(err){
-    if (err) return next(err);
-    user = group = keep = null;
-    return res.send(204);
+  Group.findOne({"_id": group._id}).populate('members').exec(function (err, group) {
```

Comment:
@paglias I'm not completely sure, but I am pretty sure.

If there is a way to populate the users into the live group object, then we don't need to.

The reason I think we need the members populates is shown in my additions to selecting the senior member. Before, we simply selected the next id that was not equal to the leader's id. @Alys stated she wanted to make sure the user had logged in recently, so I need to populate the users to access the lastCron property. Also, I believe there should be more criteria in selecting the senior member, so we will probably need more properties from the members. 

We can use projects to lighten the query though, as we don't need every property.